tests/basic_test.go

```diff
@@ -77,3 +81,98 @@ func TestJobSubmission(t *testing.T) {
 		t.Error(err.Error())
 	}
 }
+
+func TestSparkHistoryServerInstallation(t *testing.T) {
+	awsAccessKey := utils.GetenvOr("AWS_ACCESS_KEY_ID", "")
+	awsAccessSecret := utils.GetenvOr("AWS_SECRET_ACCESS_KEY", "")
+	awsBucketName := utils.GetenvOr("AWS_BUCKET_NAME", "infinity-artifacts-ci")
+	awsBucketPath := "s3a://" + awsBucketName + "/autodelete7d/kudo-spark-operator"
+
+	historyParams := make(map[string]string)
+	historyParams["enableHistoryServer"] = "true"
+	historyParams["historyServerFsLogDirectory"] = awsBucketPath
+	historyParams["historyServerOpts"] = "-Dspark.hadoop.fs.s3a.access.key=" + awsAccessKey +
+		" -Dspark.hadoop.fs.s3a.secret.key=" + awsAccessSecret +
+		" -Dspark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem"
+
+	spark := utils.SparkOperatorInstallation{
+		Params: historyParams,
+	}
+	err := spark.InstallSparkOperator()
+	defer spark.CleanUp()
+
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	awsParams := map[string]interface{}{
+		"AwsBucketPath":   awsBucketPath,
+		"AwsAccessKey":    awsAccessKey,
+		"AwsAccessSecret": awsAccessSecret,
+	}
+	job := utils.SparkJob{
+		Name:     "history-server-linear-regression",
+		Params:   awsParams,
+		Template: "spark-linear-regression-history-server-job.yaml",
+	}
+
+	// Submit a SparkApplication
+	err = spark.SubmitJob(&job)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	err = spark.WaitUntilSucceeded(job)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	// Find out History Server POD name
+	historyServerPodName, err := utils.Kubectl(
+		"get",
+		"pods",
+		"--namespace="+spark.Namespace,
+		"--output=jsonpath={.items[?(@.metadata.labels.app\\.kubernetes\\.io/name==\"history-server\")].metadata.name}",
+	)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	// Find out the Job ID for the submitted SparkApplication
+	jobId, err := utils.Kubectl(
+		"get",
+		"pods",
+		"--namespace="+spark.Namespace,
+		"--output=jsonpath={.items[*].metadata.labels.spark-app-selector}",
+	)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	// Get an application detail from History Server
+	err = utils.RetryWithTimeout(10*time.Minute, 5*time.Second, func() error {
+		historyServerResponse, err := utils.Kubectl(
+			"exec",
+			historyServerPodName,
+			"--namespace="+spark.Namespace,
+			"--",
+			"/usr/bin/curl",
+			"http://localhost:18080/api/v1/applications/"+jobId+"/jobs",
+		)
+		if err != nil {
+			return err
+		}
+
+		if len(historyServerResponse) > 0 &&
```

Comment:
nice 👍

Reply:
Thanks @samvantran for noticing. Yes History Server pod is failing, apparently Spark History Server required S3 folder to be already created. Existing folder was inside `autodelete7d` and it was deleted yesterday.